t/058-tcp-socket.t

```diff
@@ -4327,3 +4327,36 @@ failed to receive a line: closed []
 close: 1 nil
 --- no_error_log
 [error]
+
+
+
+=== TEST 72: port is optional and accept nil, default is 0
+--- config
+    server_tokens off;
+    location = /t {
+        set $port $TEST_NGINX_SERVER_PORT;
+        content_by_lua_block {
+            local sock = ngx.socket.tcp()
+            sock:settimeout(500)
+            local ok, err = sock:connect("127.0.0.1")
```

Comment:
I don't like this behavior. This may lead to hard-to-debug mistakes. When the host name part is an INET address, the port cannot be optional.

Reply:
I see, I agree with this point. So we should allow like this:
```
sock:connect("unix:/path") -- work
sock:connect("unix:/path", nil) -- work
sock:connect("unix:/path", opt) -- work

sock:connect("127.0.0.1", 80) -- work
sock:connect("127.0.0.1", 80, nil) -- work
sock:connect("127.0.0.1", 80, opt) -- work
```

But like this will be raise an error:
```
sock:connect("127.0.0.1", nil) -- not work
```

In this version, the behaviour should be like above: https://github.com/openresty/lua-nginx-module/pull/1735#discussion_r445900536, but we have to judge the domain is an unix domain or a ip.

Reply:
@rainingmaster My original comment you quoted was for the unix domain socket case, not for the INET case.

Reply:
@agentzh hihi, my quoted should be mean *not* unix domain socket?  so it should mean INET case.